src/apps/common/apputils.c

```diff
@@ -355,7 +359,7 @@ int get_raw_socket_ttl(evutil_socket_t fd, int family) {
     } while (0);
 #else
     socklen_t slen = (socklen_t)sizeof(ttl);
-    if (getsockopt(fd, IPPROTO_IPV6, IPV6_UNICAST_HOPS, &ttl, &slen) < 0) {
+    if (getsockopt(fd, IPPROTO_IPV6, IPV6_UNICAST_HOPS, (char *)&ttl, &slen) < 0) {
```

Comment:
ttl is defined as int - should it be redefined as char then?
Otherwise we may get wrong result - if value is put in the wrong byte of the int (not sure if this scenario is possible but why having this complication in the first place)

The other option is to make it `(int*)&ttl`

Reply:
Well `&ttl` is already an `int*`, so casting it like that wouldn't do anything.

Microsoft's documentation for this function ( https://learn.microsoft.com/en-us/windows/win32/winsock/ipproto-ipv6-socket-options ) says that the argument is DWORD (aka, int)

Linux man page ( https://man7.org/linux/man-pages/man7/ipv6.7.html ) says 

> Argument is a pointer to an integer. -1 in the value means use the route default, otherwise it should be between 0 and 255.

On windows, `getsockopt` explicitly takes argument of `char*` : https://learn.microsoft.com/en-us/windows/win32/api/winsock/nf-winsock-getsockopt

On POSIX (aka, linux's implementation), it's `void*` https://pubs.opengroup.org/onlinepubs/009604599/functions/getsockopt.html

The MinGW compiler explicitly uses the Microsoft API, not the POSIX one, so it's trying to use the form that takes `char*`.

```
C:/msys64/mingw64/include/winsock2.h:1010:82: note: expected 'char *' but argument is of type 'int *'
 1010 |   WINSOCK_API_LINKAGE int WSAAPI getsockopt(SOCKET s,int level,int optname,char *optval,int *optlen);
```

So my understanding of all of this is that `ttl` should be of type `int`, and casting it to `char*` is correct to support compiling on the Windows platform.

I could try casting to `void*` but i'm not sure that'll implicitly convert to `char*` on MinGW?

Reply:
> why having this complication in the first place

Yeeeeeeep, POSIX API is... yea.